helper/helper_test.go

```diff
@@ -98,4 +239,155 @@ var _ = Describe("helper tests", func() {
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []SkipFileTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
+			}
+
+			helper := NewSkipFileTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).To(BeNil())
+		})
+		It("skips batches if skip file is discovered with resize restore", func() {
+			*isResizeRestore = true
+			*origSize = 3
+			*destSize = 5
+
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []SkipFileTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
+			}
+
+			helper := NewSkipFileTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).To(BeNil())
+
+		})
+		It("skips batches if skip file is discovered with single datafile", func() {
+			*singleDataFile = true
+			*isResizeRestore = false
+			*tocFile = testTocFile
+
+			// Although pure concept would be to mock TOC file as well, to keep things simpler
+			// let's use real TOC file here
+			writeTestTOC(testTocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			oidBatch := []oidWithBatch{
+				{100, 0},
+				{200, 0},
+				{200, 1},
+				{200, 2},
+			}
+
+			expectedScenario := []SkipFileTestStep{
+				{},                               // placeholder as steps start from 1
+				{"mock_100_0", true, -1, false},  // Can open pipe for table 100, check_skip_file shall not be called
+				{"mock_200_0", true, -1, false},  // Can open pipe for table 200, check_skip_file shall not be called
+				{"mock_200_1", false, 200, true}, // Can not open pipe for table 200, check_skip_file shall called, skip file exists
+				{"mock_200_2", true, -1, false},  // Went to the next batch, Can open pipe for table 200, check_skip_file shall not be called
+			}
+
+			helper := NewSkipFileTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper, helper)
+			Expect(err).To(BeNil())
+		})
```

Comment:
The bodies of these three tests are almost identical. Can you extract the common parts into a separate function and use DescribeTable and Entry?

Reply:
Right, they look similar and perhaps moving them to separate function could save some LoCs. But this code was placed here intentionally. The idea for them is to show up the data and test scenario, so the whole test is in front of reader's eyes. The fact that lines are similar is just the coincidence.
Hiding it all under some other function would break the idea - code transparency and scenarios independence. It's still easy to amend one of them - add more steps, or change existing ones.

Reply:
But you did that in the previous [patch](https://github.com/arenadata/gpbackup/pull/103/commits/842a7037d32a77c7fd3a19605d94ff060ab7be73)!

Reply:
Different task, different objective and ideas.